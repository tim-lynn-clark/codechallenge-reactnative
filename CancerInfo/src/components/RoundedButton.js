import React from 'react';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';

/**
 * @author Tom Jay
 * @function RoundedButton
 **/
const RoundedButton = ({textValue, onPress, color, inputStyles}) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={{...styles.button, backgroundColor: color, ...inputStyles}}>
        <Text style={styles.buttonText}>{textValue}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    borderRadius: 8,
    borderColor: 'red',
    paddingVertical: 14,
    paddingHorizontal: 10,
    backgroundColor: 'orange',
    marginBottom: 10,
  },
  buttonText: {
    color: 'white',
    fontWeight: 'bold',
    fontSize: 16,
    textAlign: 'center',
    textTransform: 'uppercase',
  },
});
export default RoundedButton;
