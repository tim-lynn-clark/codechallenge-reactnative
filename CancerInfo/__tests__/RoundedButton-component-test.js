/**
 * @format
 */

import 'react-native';
import React from 'react';
import RoundedButton from '../src/components/RoundedButton';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('RoundedButton renders correctly', () => {
  const tree = renderer.create(
    <RoundedButton
      textValue="Test"
      color="blue"
      inputStyles={{width: 200, marginTop: 5}}
      onPress={() => {}}
    />,
  );
  expect(tree).toMatchSnapshot();
});
