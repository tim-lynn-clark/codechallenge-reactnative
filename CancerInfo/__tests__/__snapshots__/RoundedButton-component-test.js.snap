// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedButton renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "blue",
        "borderColor": "red",
        "borderRadius": 8,
        "marginBottom": 10,
        "marginTop": 5,
        "paddingHorizontal": 10,
        "paddingVertical": 14,
        "width": 200,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      Test
    </Text>
  </View>
</View>
`;
